File: NET/JNetCall.Sharp/Client/ServiceEnv.cs
using System;
using System.IO;

namespace JNetCall.Sharp.Client
{
    public static class ServiceEnv
    {
        private static string GetBaseFolder()
        {
            var baseDir = AppContext.BaseDirectory;
            var sep = Path.DirectorySeparatorChar;
            var parts = new[] { "net6.0", "Debug", "Release", "bin" };
            foreach (var part in parts)
            {
                var tmp = sep + part + sep;
                if (!baseDir.EndsWith(tmp))
                    continue;
                baseDir = baseDir[..^tmp.Length] + sep;
            }
            return baseDir;
        }

        public static string BuildPath(string path)
        {
            var baseDir = GetBaseFolder();
            var destDir = Path.Combine(baseDir, FixSlash(path));
            var target = Path.GetFullPath(destDir);
            return target;
        }

        private static string FixSlash(string path)
        {
            var sep = Path.DirectorySeparatorChar;
            return path.Replace('/', sep)
                .Replace('\\', sep);
        }
    }
}

File: JVM/jnetcall-java/src/main/java/jnetcall/java/client/ServiceEnv.java
package jnetcall.java.client;

import java.io.File;
import java.nio.file.Paths;

public final class ServiceEnv {

    private static String getBaseFolder() {
        try {
            var source = ServiceEnv.class.getProtectionDomain().getCodeSource();
            File jar = new File(source.getLocation().toURI().getPath());
            return jar.getParentFile().getPath();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public static String buildPath(String path) {
        var baseDir = getBaseFolder();
        var destDir = Paths.get(baseDir, fixSlash(path))
                .normalize().toAbsolutePath().toString();
        return destDir;
    }

    private static String fixSlash(String path) {
        var sep = File.separator;
        return path.replace('/' + "", sep)
                .replace('\\' + "", sep);
    }

    public static File getCurrentDir() {
        var dir = Paths.get("").toAbsolutePath();
        var txt = dir.normalize().toString();
        return new File(txt);
    }
}
